contrib/Joshua/scripts/localClusterStart.sh

```diff
@@ -6,13 +6,24 @@ LOGDIR="${WORKDIR}/log"
 ETCDIR="${WORKDIR}/etc"
 BINDIR="${BINDIR:-${SCRIPTDIR}}"
 FDBSERVERPORT="${FDBSERVERPORT:-4500}"
+SERVERCHECKS="${SERVERCHECKS:-10}"
+CONFIGUREWAIT="${CONFIGUREWAIT:-240}"
 FDBCONF="${ETCDIR}/fdb.cluster"
 LOGFILE="${LOGFILE:-${LOGDIR}/startcluster.log}"
+AUDITCLUSTER="${AUDITCLUSTER:-0}"
+AUDITLOG="${AUDITLOG:-/tmp/audit-cluster.log}"
 
 # Initialize the variables
 status=0
 messagetime=0
 messagecount=0
+let index2="${RANDOM} % 256"
+let index3="${RANDOM} % 256"
+let index4="(${RANDOM} % 255) + 1"
+# Define a random ip address on localhost
+IPADDRESS="127.${index2}.${index3}.${index4}"
```

Comment:
Can you remind me -- what is the expected model for how server processes will be run on a host? In particular, how many server processes might we be running at the same time where collisions could happen?

In the event of a collision, are we going to see a test failure?

Reply:
We can run up-to the number of CPUs Joshua agents on a machine assuming that CPU performance permits.  In a non-containerized environment each fdbserver process must listen on unique ip address/port combination. This randomization facilitates such uniqueness with pretty good success. On a 100 core machines (such as our Dev server), if all cores are running agents, we will have collisions about 100/(256*256*255) times. When a collision occurs, the server process will fail to start and the test will be marked a failure.

Reply:
If I evaluated this correctly, I think the chance of a collision given these numbers is about .03% if running 100 simultaneous tests. That sounds high enough that we might expect to see some occasional failures from this. It's not fatal to us using this method (especially since we see a higher rate of spurious errors now), but I think we should plan to eliminate this source of false positives in the future.

Reply:
The probability of it occurring is the following:
probability = 1 - ((255*255*254)! /  (((255*255*254) - 100)! * (255*255*254)^100)) ~= 0.0003 = 0.03%

I can reduce this percent even more by randomizing the port: say from the range 4000-4999

Reply:
That sounds like a good change

Reply:
done
probability:  0.03% -> 0.00003%   (very negligible)